src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -112,7 +112,7 @@ async function get_object(req, res) {
             return;
         }
     }
-
+    http_utils.set_response_override(req, res);
```

Comment:
this is called too late, as there are also returns before.
move to just below the s3_utils.set_response_object_md() - I would add in line 51.